holoviews/plotting/renderer.py

```diff
@@ -322,7 +334,10 @@ def components(self, obj, fmt=None, comm=True, **kwargs):
                                                        comm_id=plot.comm.id,
                                                        plot_id=plot_id)
                 js = '\n'.join([js, comm_js])
-            html = "<div style='display: table; margin: 0 auto;'>%s</div>" % html
+            element_id = plot_id
+            html = "<div id='%s' style='display: table; margin: 0 auto;'>%s</div>" % (element_id, html)
+        if not os.environ.get('HV_DOC_HTML', False) and js is not None:
+            js = embed_js.format(element_id=element_id, plot_id=plot_id, html=html) + js
```

Comment:
Why define ``element_id`` if you can just replace this with ``plot.id`` (or ``plot_id``)?

Reply:
Nevermind, I see it now. Though isn't it ``plot.id`` in both branches?

Reply:
Bit confusing but in the widget case they aren't identical, in that case ``plot`` variable is actually the widget and the two are different. I might just call it ``widget_id`` instead of element_id`` and then default it to None in the non-widget case.

Reply:
Ok. If it can be made clearer, then great!